server/parsers/basic_input_output_parser.ts

```diff
@@ -13,19 +13,79 @@ const sanitize = (content: string) => {
   return DOMPurify.sanitize(content, { FORBID_TAGS: ['img'] }).replace(/!+\[/g, '[');
 };
 
+const isEveryItemString = (array?: unknown) =>
+  !!(Array.isArray(array) && array.every((item) => typeof item === 'string'));
+
+export const parseSuggestedActions = (string: string): string[] => {
+  if (!string) {
+    return [];
+  }
+  const suggestedOutputString = string;
+  let suggestedActions: string[] = [];
+  try {
+    suggestedActions = JSON.parse(suggestedOutputString || '[]');
+  } catch (e) {
+    suggestedActions = [];
+  }
+
+  if (suggestedActions.length) {
+    return suggestedActions;
+  }
+
+  /**
+   * Get json-like substring from a string
+   *
+   *  /\{               // Match an opening curly brace
+   *    (?:            // Start a non-capturing group
+   *      [^{}]        // Match any character that is not a curly brace
+   *      |            // OR
+   *      (?:          // Start a non-capturing group
+   *        \{         // Match an opening curly brace
+   *        [^{}]*     // Match any characters that are not curly braces (allowing nested structures)
+   *        \}         // Match a closing curly brace
+   *      )
+   *    )*             // Repeat the non-capturing group zero or more times (allowing nested structures)
+   *  \}/g             // Match a closing curly brace, and 'g' flag for global search
+   *
+   */
+  const jsonPattern = /\{(?:[^{}]|(?:\{[^{}]*\}))*\}/g;
+
+  /**
+   * Use the regular expression to find the JSON substring
+   */
+  const match = string.match(jsonPattern);
+
+  const matchedResult = match && match[0];
+
+  if (!matchedResult) {
+    return [];
```

Comment:
Maybe add a test case to cover this line? When the output doesn't contain any json-like content.

Reply:
Done and thanks for catching.